src/utils/global-D3-tests.js

```diff
@@ -112,30 +112,40 @@ export function testToolTip(areas, toolTipDataName, areaDataName) {
     it(`My tooltip should have a "${toolTipDataName}" property that
     corresponds to the "${areaDataName}" of the active area.`,
     function() {
-       const tooltip = document.getElementById('tooltip');
-       const randomIndex = getRandomIndex(areas.length);
-       let randomArea;
-
-       assert.isNotNull(
-         tooltip.getAttribute(toolTipDataName),
-         `Could not find property "${toolTipDataName}" in tooltip `
-       );
+      const tooltip = document.getElementById('tooltip');
+      const randomIndex = getRandomIndex(areas.length);
+      let randomArea;
 
-       randomArea = areas[randomIndex];
+      assert.isNotNull(
+        tooltip.getAttribute(toolTipDataName),
+        `Could not find property "${toolTipDataName}" in tooltip `
+      );
 
-       triggerMouseEvent(randomArea, 'mouseover');
-       triggerMouseEvent(randomArea, 'mousemove');
-       triggerMouseEvent(randomArea, 'mouseenter');
+      randomArea = areas[randomIndex];
 
-       assert.equal(
-         tooltip.getAttribute(toolTipDataName),
-         randomArea.getAttribute(areaDataName),
-         `Tooltip's "${toolTipDataName}" property should be equal to the ` +
-         `active area's "${areaDataName}" property`
-       );
+      triggerMouseEvent(randomArea, 'mouseover');
+      triggerMouseEvent(randomArea, 'mousemove');
+      triggerMouseEvent(randomArea, 'mouseenter');
 
-       // Clear out tooltip.
-       triggerMouseEvent(randomArea, 'mouseout');
+      return new Promise((resolve, reject) => {
+        // Timeout is used to accommodate tooltip transitions.
+        setTimeout(() => {
+          try {
+            assert.equal(
+              tooltip.getAttribute(toolTipDataName),
+              randomArea.getAttribute(areaDataName),
+              `Tooltip's "${toolTipDataName}" property should be equal to ` +
+              `the active area's "${areaDataName}" property`
```

Comment:
It was done as the test above.
Do you think it does not work as intended?  Why do you think so?
What improvements do you suggest?

Reply:
I'm not sure if it works as intended. Promises can be a little tricky. 

Do you know how `await` works? That usually makes working with promises a little clearer. From the Mocha link I gave, I would imagine something like this would be easier to understand:

```
const toolTipText = await getTooltipDataName(tooltip);

assert.equal(toolTipText, etc.......);
```

Then you just need to put the promise related stuff inside a separate function `getTooltipDataName`;

Something like that. You could probably use better names and make the function less specific and reusable, and maybe even put it in https://github.com/freeCodeCamp/testable-projects-fcc/blob/master/src/utils/element-utils.js ?

Reply:
In this case, if we don't catch an exception, it'll be uncaught.

Let it be as is, and we will do async functions in the next PRs. Because I think, it must be done not only in this place.

Reply:
You can just add `try` and `catch` around the `await` statement.

```
try {
  const toolTipText = await getTooltipDataName(tooltip);
} catch {
  ...
}
```

Reply:
Here is more information about the advantages of `await`.

https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

Reply:
with `await`, it will not be needed. This PR addresses a concrete error with a timeout. And it resolved like other tests. I suggest redo it and other tests with async/await functionality in new PR.

Reply:
Sorry, but the code in this PR is confusing and could be written in a clearer way today. There is zero reason to avoid writing a clearer version of this code in this PR.

Reply:
@ValeraS Why was this PR closed? Are you unwilling to participate in the normal code review process and make suggested improvements? That's a very big concern when deciding whether or not to spend the time reviewing your future PRs. I feel like I just wasted a lot of time reviewing this PR if you're just going to close it after someone suggests improvements.

Reply:
IMO, you wanted new functionality. This was done in the same style as other tests. And I suggest change style in another PR. You don't agree, what I can do?

When in another PR I replace `InnerHTML` with `InnerText` you said it does not appropriate as long as everywhere we use `InnerHTML`.  Now I correct a concrete error and you suggest me change used method only in this case.

I agree that the use of async is a clearer way.

Reply:
What new functionality did I ask for? The improvements I am suggesting do not add new functionality. It just makes the code easier to read.

Reply:
A test above of this use the same code.

Reply:
> When in another PR I replace InnerHTML with InnerText you said it does not appropriate as long as everywhere we use InnerHTML. Now I correct a concrete error and you suggest me change used method only in this case.

I didn't realize the existing code already used a promise and you just copied that style. This is why we discuss things in a professional and calm way as a team instead of just closing PRs in frustration.

Since the existing code already uses a promise in the same style then I agree that we could merge this PR as-is, and then submit a separate PR to make the code clearer using `await` and `async`.

> You don't agree, what I can do?

Discuss it. Like we just did, and we came to an agreement.

Reply:
> When in another PR I replace InnerHTML with InnerText you said it does not appropriate as long as everywhere we use InnerHTML. Now I correct a concrete error and you suggest me change used method only in this case.

I didn't realize the existing code already used a promise and you just copied that style. This is why we discuss things in a professional and calm way as a team instead of just closing PRs in frustration.

Since the existing code already used a promise in the same style then I agree that we could merge this PR as-is, and then submit a separate PR to make the code clearer using await and async.

> You don't agree, what I can do?

Discuss it and try to reach an agreement, like we just did.

Reply:
sorry, I've been in a bad mood, I've restored all.